engine/verification/assigner/engine_test.go

```diff
@@ -1 +1,435 @@
+
+	// sends containerBlock containing receipt to assigner engine
+	e.ProcessFinalizedBlock(containerBlock)
+
+	mock.AssertExpectationsForObjects(t,
+		s.metrics,
+		s.assigner,
+		s.chunksQueue,
+		s.notifier,
+		s.newChunkListener,
+		s.indexer)
+}
+
+// TestChunkQueue_UnhappyPath_Error evaluates that if chunk queue returns an error upon submission of a
+// chunk to it, the new job listener is never invoked. This is important as without a new chunk successfully
+// added to the chunks queue, the consumer should not be notified.
+func TestChunkQueue_UnhappyPath_Error(t *testing.T) {
+	t.Parallel()
+
+	s := SetupTest()
+	e := NewAssignerEngine(s)
+
+	// creates a container block, with a single receipt, that contains a single chunk assigned
+	// to verification node.
+	containerBlock, assignment := createContainerBlock(
+		test.WithChunks(test.WithAssignee(s.myID())))
+	result := &containerBlock.Payload.Receipts[0].ExecutionResult
+	s.mockStateAtBlockID(result.BlockID)
+	chunksNum := s.mockChunkAssigner(result, assignment)
+	require.Equal(t, chunksNum, 1)
+
+	// mocks processing assigned chunks
+	// adding new chunks to queue results in an error
+	s.chunksQueue.On("StoreChunkLocator", mock.Anything).
+		Return(false, fmt.Errorf("error")).
+		Times(chunksNum)
+
+	// once assigner engine is done processing the block, it should notify the processing notifier.
+	s.notifier.On("Notify", containerBlock.ID()).Return().Once()
+
+	// mocks indexer module
+	// on receiving a new finalized block, indexer indexes all its receipts
+	s.indexer.On("IndexReceipts", containerBlock.ID()).Return(nil).Once()
+
+	// sends block containing receipt to assigner engine
+	e.ProcessFinalizedBlock(containerBlock)
+
+	mock.AssertExpectationsForObjects(t,
+		s.metrics,
+		s.assigner,
+		s.notifier,
+		s.chunksQueue,
+		s.indexer)
+
+	// job listener should not be notified as no new chunk is added.
+	s.newChunkListener.AssertNotCalled(t, "Check")
+}
+
+// TestChunkQueue_UnhappyPath_Duplicate evaluates that after submitting duplicate chunk to chunk queue, assigner engine does not invoke the notifier.
+// This is important as without a new chunk successfully added to the chunks queue, the consumer should not be notified.
+func TestChunkQueue_UnhappyPath_Duplicate(t *testing.T) {
+	t.Parallel()
+
+	s := SetupTest()
+	e := NewAssignerEngine(s)
+
+	// creates a container block, with a single receipt, that contains a single chunk assigned
+	// to verification node.
+	containerBlock, assignment := createContainerBlock(
+		test.WithChunks(test.WithAssignee(s.myID())))
+	result := &containerBlock.Payload.Receipts[0].ExecutionResult
+	s.mockStateAtBlockID(result.BlockID)
+	chunksNum := s.mockChunkAssigner(result, assignment)
+	require.Equal(t, chunksNum, 1)
+
+	// mocks processing assigned chunks
+	// adding new chunks to queue returns false, which means a duplicate chunk.
+	s.chunksQueue.On("StoreChunkLocator", mock.Anything).
```

Comment:
If we use the real chunks queue, then we could ask the engine to process the same block twice, and verify the chunks queue didn't store duplication.

Reply:
This is going to be addressed in https://github.com/dapperlabs/flow-go/issues/5294.